### DISCUSSION TITLE
How to handle new changes on a repository? [question]

### DISCUSSION THREAD

**Discussion by ShivangKakkar** [gitpython-developers/GitPython#1343] [Q&A] [answered]
I want to track new commits, issues, pull requests, etc of some (at least mine) github repository.
So is there a decorator like `@repo.on_new_commit()` or `@repo.on_new_issue()` or is there some other way to handle these updates in an easy fashion? I tried finding something in documentation but couldn't. 

**Please let me know how.**

**Comment by Byron** [comment, marked as answer]
No, there is no such thing as GitHub API access via GitPython.

**Reply by ShivangKakkar** [reply]
I know for sure it's possible using Github API. Do you mean GitPython can't handle any new updates (like Github sends `notifications` using emails and dashboard) on repositories? If yes, could I expect such feature in near future?

**Reply by Byron** [reply]
All GitPython does is to interact with git repositories. Since commits or PRs can be fetched with the correct ref-spec, it's possible to see them in the local clone each time such a fetch is performed. Thus it's possible to observe some of what's happening on GitHub by polling.

As GitPython is in maintenance mode, new features or enhancements are contributed, there is no roadmap for new features either.

**Reply by ShivangKakkar** [reply]
Thanks for the reply, and for the library :)
